apps/remix-ide/src/blockchain/execution-context.js

```diff
@@ -176,6 +177,7 @@ export class ExecutionContext {
         const block = await web3.eth.getBlock('latest')
         // we can't use the blockGasLimit cause the next blocks could have a lower limit : https://github.com/ethereum/remix/issues/506
         this.blockGasLimit = (block && block.gasLimit) ? Math.floor(block.gasLimit - (5 * block.gasLimit) / 1024) : this.blockGasLimitDefault
+        this.lastBlock = block
```

Comment:
`_updateChainContext` should be updated as soon as possible when a new provider is switched to.

Reply:
the https://github.com/ethereum/remix-project/pull/1412/files/4944a397f4e8222aa6c9f3ac6e1ac1247a851c79#diff-e25e2a645b7e5c5e40e6acd6a0f942ddca48aac37521c89b1e514c3f3d582c7fL146 is already there